nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -330,37 +330,27 @@ in {
         }
       ];
 
-      warnings = []
-        ++ (optional (cfg.poolConfig != null) ''
-          Using config.services.nextcloud.poolConfig is deprecated and will become unsupported in a future release.
-          Please migrate your configuration to config.services.nextcloud.poolSettings.
-        '')
-        ++ (optional (versionOlder cfg.package.version "18") ''
-          A legacy Nextcloud install (from before NixOS 20.03) may be installed.
-
-          You're currently deploying an older version of Nextcloud. This may be needed
-          since Nextcloud doesn't allow major version upgrades that skip multiple
-          versions (i.e. an upgrade from 16 is possible to 17, but not 16 to 18).
-
-          It is assumed that Nextcloud will be upgraded from version 16 to 17.
+      warnings = let
+        upgradeWarning = major: latest: nixos:
```

Comment:
Does `latest` need to be a parameter here? It doesn't seem to be used with anything other than `20`?

Reply:
I think if we have `latest` that `major` would be better understandable as `current`.

Reply:
@mweinelt what do you mean?

Reply:
AFAIU, only _if_ we keep the `latest` parameter, rename the `major` parameter to `current`. I'm for dropping the `latest` parameter though :)

Reply:
I have no strong opinion on this anyhow. The rest of the PR looks good from a cursory glance.

Reply:
@lheckemann fixed :)